### ISSUE TITLE
the initialization of batchnorm and some other layers

### ISSUE THREAD

**Issue by qinziqiao** [AlexHex7/Non-local_pytorch#22]

**Comment by qinziqiao** [comment]
```
nn.init.constant_(self.W[1].weight, 0)
nn.init.constant_(self.W[1].bias, 0)
```

I have the question why the bn.weight is initialized as zero

**Comment by AlexHex7** [comment]
@qinziqiao Hi, you can find the reason in Section 4.1 of the Paper which said

> The scale parameter of this BN layer is initialized as zero, following [17]. This ensures that the initial state of the entire non-local block is an identity mapping, so it can be inserted into any pre-trained
> networks while maintaining its initial behavior.

**Comment by qinziqiao** [comment]
Thanks for your reply. I'm so careless that neglect this line.
